data.go

```diff
@@ -0,0 +1,193 @@
+// Copyright 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package apidVerifyApiKey
+
+import (
+	"database/sql"
+	"errors"
+	"github.com/30x/apid-core"
+	"strings"
+	"sync"
+)
+
+type dbManager struct {
+	data      apid.DataService
+	db        apid.DB
+	dbMux     sync.RWMutex
+	dbVersion string
+}
+
+func (dbc *dbManager) setDbVersion(version string) {
+	db, err := dbc.data.DBVersion(version)
+	if err != nil {
+		log.Panicf("Unable to access database: %v", err)
+	}
+	dbc.dbMux.Lock()
+	dbc.db = db
+	dbc.dbMux.Unlock()
+	dbc.dbVersion = version
+	// TODO : check if we need to release old db here...
+}
+
+func (dbc *dbManager) getDb() apid.DB {
+	dbc.dbMux.RLock()
+	defer dbc.dbMux.RUnlock()
+	return dbc.db
+}
+
+func (dbc *dbManager) initDb() error {
+	db := dbc.getDb()
+	if db == nil {
+		return errors.New("DB not initialized")
+	}
+	return nil
+}
+
+func (dbc *dbManager) getDbVersion() string {
+	return dbc.dbVersion
+}
+
+type dbManagerInterface interface {
+	setDbVersion(string)
+	initDb() error
+	getDb() apid.DB
+	getDbVersion() string
+	getKmsAttributes(tenantId string, entities ...string) map[string][]Attribute
+	getApiKeyDetails(dataWrapper *VerifyApiKeyRequestResponseDataWrapper) error
+}
+
+func (dbc *dbManager) getKmsAttributes(tenantId string, entities ...string) map[string][]Attribute {
+
+	db := dbc.db
+	var attName, attValue sql.NullString
+	var entity_id string
```

Comment:
I would suggest use `sql.NullString`, though the column is not nullable. An error in db row should not panic our apid.